sqlglot/dialects/exasol.py

```diff
@@ -0,0 +1,453 @@
+        "YY": "%y",  # Last 2 digits of year (e.g., 25)
+        "Y": "%y",  # Last digit of year
+        # --- Month ---
+        "MM": "%m",  # Month (01-12)
+        "MON": "%b",  # Abbreviated month name (e.g., JAN)
+        "MONTH": "%B",  # Full month name (e.g., JANUARY)
+        "RM": "%m",  # Roman numeral month (no direct strftime, maps to decimal)
+        # --- Day ---
+        "DD": "%d",  # Day of month (01-31)
+        "DDD": "%j",  # Day of year (001-366)
+        "DY": "%a",  # Abbreviated weekday name (e.g., MON)
+        "DAY": "%A",  # Full weekday name (e.g., MONDAY)
+        "D": "%w",  # Day of week (1-7, 1=Sunday, matches %w; for ISO 8601 day, see ID)
+        # --- Hour ---
+        "HH24": "%H",  # Hour (00-23)
+        "HH12": "%I",  # Hour (01-12)
+        "HH": "%I",  # (alias for HH12)
+        "AM": "%p",  # AM/PM meridian indicator
+        "PM": "%p",  # AM/PM meridian indicator
+        # --- Minute ---
+        "MI": "%M",  # Minute (00-59)
+        # --- Second ---
+        "SS": "%S",  # Second (00-59)
+        # Fractional Seconds: Exasol uses FF[1-9]. strftime uses %f (microseconds)
+        "FF": "%f",  # Maps to microseconds. Precision (FF1-FF9) might need truncation/padding.
+        "FF1": "%f",
+        "FF2": "%f",
+        "FF3": "%f",  # Often used for milliseconds
+        "FF4": "%f",
+        "FF5": "%f",
+        "FF6": "%f",  # Microseconds
+        "FF7": "%f",
+        "FF8": "%f",
+        "FF9": "%f",
+        # --- Timezone ---
+        "TZH": "%z",  # Timezone hour offset (+/-HH)
+        "TZM": "%z",  # Timezone minute offset (often combined with TZH)
+        "TZHTZM": "%z",  # Combined timezone offset (+/-HHMM). Note: strftime %z is +/-HHMM or +/-HH:MM
+        # --- ISO 8601 Specific ---
+        "IYYY": "%G",  # ISO Year (4-digit)
+        "IW": "%V",  # ISO Week of year (01-53)
+        "ID": "%u",  # ISO Day of week (1-7, 1=Monday)
+    }
+
+    class Tokenizer(tokens.Tokenizer):
+        SINGLE_TOKENS = {
+            **tokens.Tokenizer.SINGLE_TOKENS,
+        }
+        SINGLE_TOKENS.pop("%") 
+        KEYWORDS = {
+            **tokens.Tokenizer.KEYWORDS,
+        }
+        KEYWORDS.pop("DIV")
+
+    class Parser(parser.Parser):
+        # ADD_(DAYS | HOURS | MINUTES | MONTHS | SECONDS | WEEKS | YEARS)
+        # https://docs.exasol.com/db/latest/sql_references/functions/scalarfunctions.htm
+        DATE_ADD_FUNCTIONS = {
+            f"ADD_{unit}S": build_date_delta(exp.DateAdd, None, unit)
+            for unit in ["DAY", "HOUR", "MINUTE", "MONTH", "SECOND", "WEEK", "YEAR"]
+        }
+
+        NUMERIC_FUNCTIONS = {
+            "DIV": binary_from_function(exp.IntDiv),
+            "RANDOM": lambda args: exp.Rand(
+                lower=seq_get(args, 0), upper=seq_get(args, 1)
+            ),
+        }
+
+        STRING_FUNCTIONS = {
+            "ASCII": exp.Unicode.from_arg_list,
+            "REGEXP_SUBSTR": exp.RegexpExtract.from_arg_list,
```

Comment:
This looks incorrect: `RegexpExtract` doesn't represent a "regexp substring" operation ([docs](https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/regexp_substr.htm#microcontent1)).